File: Data Structures & Algorithms/duplicate-integer/submission-4.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        map<int, int> dup;
        for(int i = 0; i < nums.size(); ++i){
            cout << nums[i] << "\n";
            if (dup.count(nums[i])){
                return true;
            }
            dup[nums[i]] = 1;
        }
        return false;
    }
};

File: Data Structures & Algorithms/two-integer-sum/submission-4.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        m = defaultdict(int)

        for i in range( len(nums) ):
            if nums[i] in m.keys():
                return [ m[nums[i]], i ]
            else: 
                m[ target - nums[i]] = i
            #print(m)
        return []